Add authorization and some additional "getting started" helps

pettys opened the pull request:
- I documented a few things that tripped me up when getting the app running in a new environment.
- I added an Azure DevOps CI pipeline.
- I noticed that when a student logs in, all info for other students and all administrators was being downloaded to the student's browser. I added authorization to restrict this.

pettys committed:
Fixes/changes to help setup on new server

Commit:
Set up CI with Azure Pipelines

pettys committed:
Fix hard-coded student ID in history query

pettys committed:
Add first-pass authorization checks

pettys committed:
Improve nav bar's vertical space usage

pettys committed:
Improve student usability on MakeEntry screen

Before, it wasn't clear which pay period a student was creating an
entry for. Now the student can see the status of all the entry status of
all the recent pay periods and select which one they want to create new
entries for.

This commit also fixes the issue where s "Save Success" message would be displayed
before the save actually succeeded. Indeed, if the save failed, the student
would still see the success message. Now the form is disabled while saving is
in progress, and the success message is only shown on actual success.

pettys commented:
@cneisen234 @KyleHagler-bit @Learn-Or-Teach @Dalton-Meyer @clevereda

Hi guys - I've been working on this project a bit, and the Foundation is getting ready to roll things out to live students in a week or so here, so that's pretty cool!

Since you haven't accepted this pull request, I'll take it that you're all on to whatever cool stuff you're finding to do after your EDA experience, and I won't submit any more with future changes. Just wanted to give you a shout-out that the Foundation is taking what you've started and building on it, so congratulations on that, and feel free to reach out any time. Good luck!

Jason

(cc @booherbg)

cneisen234 commented:
Thanks for the update Jason. I’m happy to hear that they will benefit from
this.

On Tue, Nov 24, 2020 at 11:50 AM Jason Pettys <notifications@github.com>
wrote:

> @cneisen234 <https://github.com/cneisen234> @KyleHagler-bit
> <https://github.com/KyleHagler-bit> @Learn-Or-Teach
> <https://github.com/Learn-Or-Teach> @Dalton-Meyer
> <https://github.com/Dalton-Meyer> @clevereda
> <https://github.com/clevereda>
>
> Hi guys - I've been working on this project a bit, and the Foundation is
> getting ready to roll things out to live students in a week or so here, so
> that's pretty cool!
>
> Since you haven't accepted this pull request, I'll take it that you're all
> on to whatever cool stuff you're finding to do after your EDA experience,
> and I won't submit any more with future changes. Just wanted to give you a
> shout-out that the Foundation is taking what you've started and building on
> it, so congratulations on that, and feel free to reach out any time. Good
> luck!
>
> Jason
>
> (cc @booherbg <https://github.com/booherbg>)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Team-Legacy-Foundation/legacy_foundation/pull/194#issuecomment-733137970>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANNXSNZLC4KJJV2B5GCZJA3SRPW77ANCNFSM4TSLSMXA>
> .
>

pettys committed:
Clear up some code warnings; add some readme help

pettys committed:
Don't allow student entry after pay day deadline

pettys committed:
Allow admins to edit student entries

pettys committed:
Fix entry ID getting squashed by student ID

pettys committed:
Update entries by their ID (see notes)

Before we were updating by LCF ID, so updating any
entry would update all entry for that student.

Also added pay day to admin entry list, since the
table was already showing all entries regardless
of the pay_day value. This was confusingly
presented and is now clearer.

pettys committed:
Fix a bad 'where' clause
